src/appsync.ts

```diff
@@ -180,62 +198,36 @@ export class AppsyncResolver<
     ResolverFunction<Arguments, Result, Source>
   >;
 
-  constructor(fn: ResolverFunction<Arguments, Result, Source>) {
-    this.decl = validateFunctionDecl(fn, "AppsyncResolver");
-  }
+  public readonly resource;
+  public readonly resolvers;
 
-  /**
-   * Generate and add an AWS Appsync Resolver to an AWS Appsync GraphQL API.
-   *
-   * ```ts
-   * import * as appsync from "@aws-cdk/aws-appsync-alpha";
-   *
-   * const api = new appsync.GraphQLApi(..);
-   *
-   * ```ts
-   * const getPerson = new AppsyncResolver<{id: string}, Person | undefined>(
-   *   ($context, id) => {
-   *     const person = table.get({
-   *       key: {
-   *         id: $util.toDynamoDB(id)
-   *       }
-   *     });
-   *     return person;
-   *   });
-   * ```
-   *
-   * getPerson.createResolver(api, {
-   *   typeName: "Query",
-   *   fieldName: "getPerson"
-   * });
-   * ```
-   *
-   * @param api the AWS Appsync API to add this Resolver to
-   * @param options typeName, fieldName and cachingConfig for this Resolver.
-   * @returns a reference to the generated {@link appsync.Resolver}.
-   */
-  public addResolver(
-    api: appsync.GraphqlApi,
-    options: Pick<
-      appsync.BaseResolverProps,
-      "typeName" | "fieldName" | "cachingConfig"
-    >
-  ): SynthesizedAppsyncResolver {
-    const fields = this.getResolvableFields(api);
-
-    return new SynthesizedAppsyncResolver(
-      api,
-      `${options.typeName}${options.fieldName}Resolver`,
-      {
-        ...options,
-        api,
-        templates: fields.templates,
-        dataSource: fields.dataSource,
-        requestMappingTemplate: fields.requestMappingTemplate,
-        responseMappingTemplate: fields.responseMappingTemplate,
-        pipelineConfig: fields.pipelineConfig,
-      }
-    );
+  constructor(
+    scope: Construct,
+    id: string,
+    props: AppsyncResolverProps<Arguments, Result, Source>
+  ) {
```

Comment:
Could the scope come from the api?

Reply:
That would force all resolvers to exist in the same stack

Reply:
I mean as an option, with event bus many of the methods have the scope as optional with the bus being used when not provided. obv just a nice to have.